flight/Modules/Stabilization/stabilization.c

```diff
@@ -312,16 +314,18 @@ static void stabilizationTask(void* parameters)
 			switch(stabDesired.StabilizationMode[i])
 			{
 				case STABILIZATIONDESIRED_STABILIZATIONMODE_RATE:
-					if(reinit)
-						pids[PID_RATE_ROLL + i].iAccumulator = 0;
-
-					// Store to rate desired variable for storing to UAVO
-					rateDesiredAxis[i] = bound_sym(stabDesiredAxis[i], settings.ManualRate[i]);
-
-					// Compute the inner loop
-					actuatorDesiredAxis[i] = pid_apply_setpoint(&pids[PID_RATE_ROLL + i],  rateDesiredAxis[i],  gyro_filtered[i], dT);
-					actuatorDesiredAxis[i] = bound_sym(actuatorDesiredAxis[i],1.0f);
+					// Store for debugging output TODO(fujin): Why?
+					rateDesiredAxis[i] = stabDesiredAxis[i];
 
+					// Run a gyro rate stabilization algorithm on this axis
+					stabilization_rate(gyro_filtered[i],
+                             rateDesiredAxis[i],
+                             &actuatorDesiredAxis[i],
+                             dT,
+                             reinit,
+                             i,
+                             &pids[PID_RATE_ROLL + i],
+                             &settings);
```

Comment:
Yeah but the indenting is all broken due to spreading it over multiple lines.. will take another pass at it